weave/trace_server/tests/test_orm.py

```diff
@@ -173,3 +174,117 @@ def test_select_fields():
     )
     assert prepared.parameters == {"pb_0": '$."address"'}
     assert prepared.fields == ["id", "payload.address"]
+
+
+def test_join():
+    table1 = Table(
+        "users",
+        [
+            Column("id", "string"),
+            Column("creator", "string", nullable=True),
+            Column("payload", "json", db_name="payload_dump"),
+        ],
+    )
+    table2 = Table(
+        "roles",
+        [
+            Column("id", "string"),
+            Column("name", "string"),
+        ],
+    )
+
+    select = (
+        table1.select()
+        .fields(["id", "name"])
+        .join(
+            "",
+            table2,
+            tsi.Query(
```

Comment:
I know we want to guard against SQL injection, but it feels wordy and weird to me to use this MongoDB syntax for the join condition.

Reply:
I too think its wordy, but i feel like having to parse which conditions require the mongo syntax and which dont feels annoying as well, vs everything in the mongo syntax